app/src/main/java/mozilla/lockbox/presenter/EditItemPresenter.kt

```diff
@@ -146,19 +153,29 @@ class EditItemPresenter(
             .addTo(compositeDisposable)
 
         view.saveEntryClicks
-            .subscribe {
-                credentials?.let {
-                    dispatcher.dispatch(DataStoreAction.UpdateItemDetail(it))
-                    view.closeKeyboard()
-                    dispatcher.dispatch(RouteAction.ItemList)
-                } ?: pushError(
-                    NullPointerException("Credentials are null"),
-                    "Error editing credential with id ${credentials?.id}"
-                )
+            .map {
+                edited?.let { ItemDetailAction.SaveChanges(it) }
+                    ?: ItemDetailAction.DiscardChanges(itemId)
             }
+            .subscribe(dispatcher::dispatch)
             .addTo(compositeDisposable)
     }
 
+    private fun checkDismissChanges(itemId: String) =
+        edited?.let {
+            if (it != credentials) {
+                DialogAction.DiscardChangesDialog(itemId)
+            } else {
+                null
```

Comment:
This could be written as

```kotlin
// if the user has changed the credentials
if (edited != null && edited != credentials) {
   // display a dialog checking if they want to discard
   DialogAction.DiscardChangesDialog(itemId)
} else {
   // just go ahead and discard/exit.
   ItemDetailAction.DiscardChanges(itemId)
}
```